library/agent/Config.ts

```diff
@@ -14,8 +16,12 @@ export type Endpoint = {
   };
 };
 
+export type Endpoint = Omit<EndpointConfig, "allowedIPAddresses"> & {
+  allowedIPAddresses?: IPMatcher;
```

Comment:
Took me a second to know why we use `Omit` but otherwise `allowedIPAddresses` becomes `never`, right?

```suggestion
  allowedIPAddresses: IPMatcher | undefined;
```

I think I might prefer setting it explicitly to `undefined` (now the property key doesn't need to exist)?

Reply:
The issue is that it still allows `allowedIPAddresses` to be an array then 😬. But I will change to explicit undefined.